### ISSUE TITLE
Play position reverted on bluetooth disconnect

### ISSUE THREAD

**Issue by Redsandro** [flackbash/AudioAnchor#158]
### Problem

I notice AudioAnchor forgetting the most recent play position intermittently. I think there are multiple ways to trigger this, but the one reproducible way I found was to stop playback by turning off a Bluetooth headset, and at a later time, reconnect and press play. It will play from the position it was prior to listening.

When listening to a really good book, I have to manually seek to the point I was last listening daily. 

It also seems to happen (intermittently?) when pressing pause on the Bluetooth headset prior to turning it off.

### Hypothesis

AudioAnchor does not store the latest playback position when the playback is stopped by powering off a Bluetooth headset/speaker.

### Workaround

The workaround is to manually grab the phone out of your luggage, unlock, open AudioAnchor, and pause from there. It defeats the purpose of Bluetooth, so I prefer triggering the issue and manually seeking later, but it becomes tedious if the book has one big file in stead of individual chapter files. :sweat_smile: 

### Other considerations

In order to prevent any similar unforeseen variant of this issue (e.g. phone powering off due to battery), it would be helpful if AudioAnchor would store the latest playback position every 10 seconds or every minute, or whatever is a sane interval.

If someone with the same use case (Bluetooth, pause by power off) does _not_ recognize this issue, then it may be some interaction between the way AudioAnchor works and the phone's power management, e.g. killing the app before the state is saved. :thinking: Although I'm using AntennaPod for podcasts, and it works without these issues.

### Similar issues found through search

- #34 
- #159 (new)

### TL;DR:

- [ ] Save playback position upon Bluetooth device disconnected.
- [ ] Save playback position intermittently.
- [ ] Save playback position on [ACTION_SHUTDOWN](https://developer.android.com/reference/android/content/Intent#ACTION_SHUTDOWN) (added from #159)
- [ ] Save playback position [onPause()](https://developer.android.com/reference/android/app/Activity#onPause()) (added from #159)

**Comment by flackbash** [comment]
I can reproduce this. Interestingly, progress seems to be lost only if the playback is started again from the Bluetooth headset/speaker. When pressing play in the app, it works fine.

Thanks for reporting it, I'll have a look.

**Comment by flackbash** [comment]
Implemented fix for this problem: Instead of saving the playback position only on static events like pause/skip/... the playback position is now saved periodically (currently every 15s) such that at most 15s of progress are lost when an unexpected process death happens.
